Sources/Kafka/KafkaHeader.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-client open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-client project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-client project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+/// A structure representing a header for a Kafka message.
+/// Headers are key-value pairs that can be attached to Kafka messages to provide additional metadata.
+public struct KafkaHeader: Sendable, Hashable {
+    /// The key associated with the header.
+    public var key: String
+
+    /// The value associated with the header.
+    public var value: ByteBuffer? // TODO: KafkaContiguousBytes
```

Comment:
Ideally, we would do the same `KafkaContiguousBytes` dance here as in `KafkaProducerMessage`.
However, the problem is that `KafkaHeader` is used in an array, which means the entire array would have to be generic over one specific `KafkaContiguousBytes` type which feels wrong too.

Reply:
I am fine to leave this with `ByteBuffer` for now. Restricting the Array to single type isn't great.